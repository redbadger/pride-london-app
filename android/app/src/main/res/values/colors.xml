<?xml version="1.0" encoding="utf-8"?>
<resources>
    <color name="splash_bg">#ffffff</color>
    <color name="primary_dark">#ffffff</color>
</resources>