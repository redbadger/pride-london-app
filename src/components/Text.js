// @flow
import React from "react";
import { Text as RnText, StyleSheet, PixelRatio } from "react-native";
import type { TextProps } from "react-native/Libraries/Text/TextProps";
import type { TextStyleProp } from "react-native/Libraries/StyleSheet/StyleSheet";
import Markdown from "react-native-easy-markdown";
import { mergeDeepRight } from "ramda";
import {
  blackColor,
  lightNavyBlueColor,
  whiteColor
} from "../constants/colors";

export type TextType =
  | "uber"
  | "h1"
  | "h2"
  | "h3"
  | "h4"
  | "text"
  | "small"
  | "xSmall"
  | "price"
  | "tabBarItem";

export type ColorType = "lightNavyBlueColor" | "blackColor" | "whiteColor";

type Props = {
  type: TextType,
  color: ColorType,
  markdown: boolean,
  markdownStyle: TextStyleProp,
  ...TextProps
};

class Text extends React.PureComponent<Props> {
  static defaultProps = {
    type: "text",
    markdown: false,
    color: "blackColor",
    markdownStyle: {}
  };

  render() {
    const {
      color,
      type,
      markdown,
      style,
      markdownStyle,
      ...otherProps
    } = this.props;
    const typedType: TextType = (type: any);
    const typedColor: ColorType = (color: any);
    return markdown ? (
      <Markdown
        style={style}
        markdownStyles={mergeDeepRight(
          mergeDeepRight(textStyles, markdownDefaultStyles),
          markdownStyle
        )}
        renderImage={() => null}
        {...otherProps}
      />
    ) : (
      <RnText
        style={[styles[typedType], styles[typedColor], style]}
        {...otherProps}
      />
    );
  }
}

const fontScaleCaps: { [TextType]: number } = {
  uber: 1,
  h1: 1,
  h2: 22 / 18,
  h3: 20 / 16,
  h4: 20 / 16,
  text: 20 / 16,
  small: 18 / 14,
  xSmall: 14 / 12,
  price: 18 / 14,
  tabBarItem: 13 / 12
};

/**
 * React Native scales font (fontSize and lineHeight) automatically based on
 * the user selected scale (retrievable by PixelRatio.getFontScale()).
 *
 * This function ensures, that the font can not scale beyond the specified
 * maximum scale. This is achieved by returning a proportial recuded font size
 * when the scale is above the specified cap.
 */
export const scaleFont = (type: TextType, base: number) => {
  const fontScale = PixelRatio.getFontScale();
  const fontScaleCap = fontScaleCaps[type];
  if (fontScale > fontScaleCap) {
    const max = base * fontScaleCap;
    return max / fontScale;
  }

  return base;
};

/**
 * All other style props (except fontSize and lineHeight) are not affected by
 * the user selected font scale. However, sometimes it is helpful to scale a
 * view based on the text size of its children.
 *
 * This function returns a size scaled with the user selected font scale but
 * capped by the maximum scale of the specified text type.
 */
export const scaleWithFont = (type: TextType, base: number) => {
  const fontScale = PixelRatio.getFontScale();
  const fontScaleCap = fontScaleCaps[type];
  return base * Math.min(fontScale, fontScaleCap);
};

const textStyles = {
  uber: {
    fontFamily: "Poppins-ExtraBold",
    fontSize: scaleFont("uber", 32),
    lineHeight: scaleFont("uber", 36),
    includeFontPadding: false
  },
  h1: {
    fontFamily: "Poppins-Bold",
    fontSize: scaleFont("h1", 24),
    lineHeight: scaleFont("h1", 28),
    includeFontPadding: false
  },
  h2: {
    fontFamily: "Poppins-SemiBold",
    fontSize: scaleFont("h2", 18),
    lineHeight: scaleFont("h2", 24),
    includeFontPadding: false
  },
  h3: {
    fontFamily: "Poppins-SemiBold",
    fontSize: scaleFont("h3", 16),
    lineHeight: scaleFont("h3", 20),
    includeFontPadding: false
  },
  h4: {
    fontFamily: "Roboto-Medium",
    fontSize: scaleFont("h4", 16),
    lineHeight: scaleFont("h4", 24)
  },
  text: {
    fontFamily: "Roboto",
    fontSize: scaleFont("text", 16),
    lineHeight: scaleFont("text", 24)
  },
  small: {
    fontFamily: "Roboto",
    fontSize: scaleFont("small", 14),
    lineHeight: scaleFont("small", 20)
  },
  xSmall: {
    fontFamily: "Poppins-SemiBold",
    fontSize: scaleFont("xSmall", 12),
    lineHeight: scaleFont("xSmall", 16),
    includeFontPadding: false
  },
  price: {
    fontFamily: "Roboto-Bold",
    fontSize: scaleFont("price", 14),
    lineHeight: scaleFont("price", 20)
  },
  tabBarItem: {
    fontFamily: "Poppins-Bold",
    fontSize: scaleFont("tabBarItem", 12),
    lineHeight: scaleFont("tabBarItem", 16),
    includeFontPadding: false
  },
  blackColor: {
    color: blackColor
  },
  lightNavyBlueColor: {
    color: lightNavyBlueColor
  },
  whiteColor: {
    color: whiteColor
  }
};

const markdownDefaultStyles = {
  // "u" - underline is listed in react-native-easy-markdown
  // but doesn't exist in the markdown spec so should be rendered bold
  u: {
    fontWeight: "bold"
  },
  list: {
    marginBottom: 5,
    marginHorizontal: 16
  },
  listItem: {
    flexDirection: "row",
    justifyContent: "flex-start",
    alignItems: "flex-start",
    marginBottom: 5
  },
  listItemBullet: {
    alignSelf: "flex-start",
    width: 4,
    height: 4,
    backgroundColor: "black",
    borderRadius: 2,
    marginTop: scaleWithFont("small", 10),
    marginRight: 10
  },
  text: {
    color: blackColor
  }
};

const styles = StyleSheet.create(textStyles);

export default Text;
