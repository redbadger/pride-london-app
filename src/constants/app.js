// @flow
export const appNameIos = "pride-in-london";
export const appIdIos = "1250496471";
