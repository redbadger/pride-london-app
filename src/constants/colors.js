export const bgColor = "rgb(255, 255, 255)";
export const cardBgColor = "rgb(255, 255, 255)";
export const imageBgColor = "#C1C1C1";
export const headerBgColor = "#A5A5A5";

export const textColor = "#272727";

export const dialogBackdropColor = "rgba(0, 0, 0, 0.7)";
export const dialogBgColor = "#FFFFFF";
export const dialogApplyButtonBgColor = "rgb(51, 51, 51)";
export const dialogApplyButtonTextColor = "#FFFFFF";
export const dialogHeaderDividerColor = "#F2F2F2";

// FILTER HEADER

export const filterBgColor = "rgb(45, 47, 127)";
export const interestButtonBgColor = "rgb(32, 33, 90)";
export const interestButtonTextColor = "rgb(44, 218, 157)";
export const filterButtonBorderColor = "rgb(44, 218, 157)";
export const filterButtonTextColor = "rgb(44, 218, 157)";

// EVENT LIST

export const eventListHeaderBgColor = "rgb(216, 216, 216)";
export const eventListHeaderColor = "rgb(51, 51, 51)";
export const eventCardTextColor = "rgb(45, 47, 127)";
export const eventPriceColor = "rgb(255, 255, 255)";
export const eventCardShadow = "rgba(0, 0, 0, 0.2)";
export const eventPriceBgColor = "rgb(45, 47, 127)";
export const sectionHeaderBgColor = "rgb(243, 243, 243)";
export const sectionHeaderShadow = "rgba(0, 0, 0, 0.14)";

// EVENT DETAILS
export const eventDetailsBgColor = "#FFF";
export const eventDetailsHeaderBgColor = "#F2F2F2";
export const eventCategoryLabelBgColor = "#D8D8D8";
export const eventDetailsBuyButtonColor = "#D8D8D8";
export const eventDetailsBuyButtonTextColor = "#FFFFFF";
