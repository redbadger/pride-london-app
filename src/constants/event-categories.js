// Ideally the keys here should be consts, but currently these are setup in
// contentful only as the label, hence we need to use this for the key too.
// If we move to realise categories as its own content type then this will
// go away and we will fetch this data.

const categories = {
  "Cabaret and Variety": {
    label: "Cabaret and Variety"
  },
  Community: {
    label: "Community"
  },
  "Talks and Debates": {
    label: "Talks and Debates"
  },
  "Film and Screenings": {
    label: "Film and Screenings"
  },
  "Plays and Theatre": {
    label: "Plays and Theatre"
  },
  "Social and Networking": {
    label: "Social and Networking"
  },
  Nightlife: {
    label: "Nightlife"
  },
  "Exhibition and Tours": {
    label: "Exhibition and Tours"
  },
  "Sports and Activities": {
    label: "Sports and Activities"
  },
  Health: {
    label: "Health"
  },
  Music: {
    label: "Music"
  }
};

export default categories;
