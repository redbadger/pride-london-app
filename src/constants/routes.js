export const EVENT_DETAILS = "EVENT_DETAILS";
export const EVENT_LIST = "EVENT_LIST";
export const FEATURED_EVENT_LIST = "FEATURED_EVENT_LIST";
export const HOME = "HOME";
export const EVENT_CATEGORIES_FILTER = "EVENT_CATEGORIES_FILTER";
export const PARADE = "PARADE";
export const SAVED = "SAVED";
export const SUPPORT_US = "SUPPORT_US";
export const FILTER_MODAL = "FILTER_MODAL";
export const DONATE = "DONATE";
export const SPONSOR = "SPONSOR";
