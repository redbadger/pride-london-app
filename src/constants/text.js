// @flow
const showEvents = (numberOfEvents: number) => {
  if (numberOfEvents <= 0) {
    return "No events";
  }

  if (numberOfEvents === 1) {
    return "Show 1 event";
  }

  return `Show ${numberOfEvents} events`;
};

const showEventsLabel = (numberOfEvents: number) => {
  if (numberOfEvents <= 0) {
    return "Show events (none selected)";
  }

  if (numberOfEvents === 1) {
    return "Show 1 selected event";
  }

  return `Show ${numberOfEvents} selected events`;
};

export const calendarTitleLabel = (dateRange: string, incomplete: boolean) =>
  `Selected: ${dateRange}${
    incomplete ? ", pick another day to select range" : ""
  }`;

export default {
  backButtonAccessibilityLabel: "Back",
  tabHome: "Home",
  tabEvents: "Events",
  tabParade: "Parade",
  tabSaved: "Saved",
  tabSupportUs: "Support us",
  tabParadeMap: "Map",
  tabParadeGroups: "Parade groups",
  featuredEventListTitle: "Featured events",
  filterByInterest: "All event types",
  categoryFilterButton: "Filter by event type",
  categoryFilterContents: "Filtering event type by",
  categoryFilterEmpty: "0 event types selected",
  addFilters: "Filter by area, price...",
  filters: "Filters",
  filterPickerApply: showEvents,
  filterPickerApplyLabel: showEventsLabel,
  filterDayPickerTitle: "Select dates",
  filterTimePickerTitle: "Select time",
  filterTitle: "Showing",
  eventDetailsPrice: "From: £",
  isFreePrice: "Free",
  eventFromPrice: "From",
  eventDetailsAccessibility: "Accessibility",
  eventDetailsAccessibilityDetails: "Accessibility Details",
  eventDetailsContact: "Contact",
  eventDetailsBuyButton: "Get tickets",
  eventDetailsGenderNeutralToilets: "Gender neutral toilets",
  selectDates: "Select dates",
  eventDetailsAbout: "About",
  eventDetailsSchedule: "Schedule",
  eventDetailsReadMore: "Read more",
  eventDetailsReadLess: "Read less",
  eventDetailsContactEmailSubject: "Pride In London event enquiry",
  eventDetailsContactEmailBody: "Dear event organiser,",
  collapsibleShowMore: "Show more",
  collapsibleShowLess: "Show less",
  savedEventsTitle: "Saved Events",
  supportUsTitle: "Support us",
  supportUsAsIndividual: "As an individual",
  supportUsAsBusiness: "As a business",
  supportUsDonate: "Donate",
  supportUsDonateDescription: "Every donation keeps us marching",
  supportUsShop: "Buy merchandise",
  supportUsShopDescription: "Visit our online shop on TheGayShop",
  supportUsSponsor: "Partner with us",
  supportUsSponsorDescription: "Support the community",
  supportUsVolunteer: "Volunteer",
  supportUsVolunteerDescription: "Passionate about equality? Join us.",
  sponsorTitle: "Partner with us",
  sponsorLevelHeading: "partners",
  sponsorHeadingOurPartners: "Our main 2018 partners",
  sponsorTextOurPartners:
    "A huge thank you to our main partners for their continued support.",
  sponsorHeadingPartnerWithUs: "Partner with us",
  sponsorTextPartnerWithUs:
    "Build your brand, engage your internal network and be part of a global movement to support diversity. To find out more, please contact sponsor@prideinlondon.org.",
  sponsorTextPartnerWithUsList:
    "* Pride in London is one of the top three biggest events in London, attracting an estimated one million visitors across the footprint. \n* Be part of our award winning marketing and advertising campaign. \n* 80% of those surveyed in our 2016 research agreed that they are more likely to think better of a brand if it is supportive of LGBT+ issues.",
  sponsorContactEmailSubject: "Sponsorship enquiry",
  sponsorContactButtonText: "Email to partner with us",
  donateTitle: "Donate",
  donateHeader: ["\"It's the first time I feel", 'accepted in a big group."'],
  donateIntroductionHeading: "Be part of the movement that drives diversity",
  donateIntroduction:
    "The need for Pride is as strong as ever. A huge amount of progress has been made for LGBT+ people but challenges and stigma are still rife. Whether you take part to celebrate, party or protest. Pride in London is your platform to make change. Your donations keep us marching.",
  donateAmountSelectionLabel: "Make a single donation",
  donateOtherAmountLabel: "Other amount (£)",
  donateMinimumAmount: "The minimum donation amount is £2.",
  donateButtonText: "Donate",
  anyDay: "Any day",
  anyTime: "Any time",
  time: {
    morning: "Morning",
    afternoon: "Afternoon",
    evening: "Evening"
  },
  tags: {
    audience: "Age group",
    price: "Price",
    free: "Show only free events",
    timeOfDay: "Time of day",
    morning: "Morning",
    afternoon: "Afternoon",
    evening: "Evening",
    venueDetails: "Venue options",
    accessibilityOptions: "Accessibility",
    "Gender neutral toilets": "Gender-neutral toilets",
    Indoors: "Indoor",
    Outdoors: "Outdoor",
    "Accessible Toilets": "Accessible toilets",
    "Wheelchair accessibile": "Wheelchair access",
    area: "Area of London"
  },
  cancel: "Cancel",
  allEvents: "All events",
  showEvents,
  showAllEvents: "Show all events",
  clearAll: "Clear all",
  reset: "Reset",
  filterEvents: "Filter Events",
  runsFrom: "Runs from:",
  noEventsTitle: "No events to show",
  noEventsDetails:
    "Try changing your filters to be less specific or check back later.",
  noSavedEventsTitle: "Where's the love?",
  noSavedEventsPart1: "No favourites yet? Tap the",
  noSavedEventsSaveButtonAlt: "'save this event' button",
  noSavedEventsPart2: "on any event you like to save them here.",
  noSavedEventsButton: "Find Events",
  saveEventButtonUnSaveEvent: "remove event from saved list",
  saveEventButtonSaveEvent: "save this event",
  homeViewAll: "View all",
  homeSupportUs: "Support us",
  homeSupportUsDescription: "Be part of the movement",
  parade: {
    headerTitle: "Parade - 7 July from 12pm"
  },
  paradeGroups: {
    title: "London Parade",
    subTitle: "Saturday 7 July from 12pm",
    paragraph1:
      "The Pride in London Parade is a platform for every part of London’s LGBT+ community. These groups, nearly 500 in total, represent our most diverse Pride in London yet. From LGBT+ parent groups and student services to sports clubs and national charities, we are proud to give a platform to our community representatives, supporters and allies.",
    paragraph2:
      "Find out who you can expect to see on the day from the list below.",
    linkLabelFacebook: "Facebook Page",
    linkLabelTwitter: "Twitter Profile",
    linkLabelWebsite: "Website",
    expandAccessibilityLabel: "Show more",
    collapseAccessibilityLabel: "Show less"
  },
  paradeInformationScreen: {
    pageHeading: "London Parade",
    pageSubheading: "Saturday 7 July",
    pageDescription:
      "The Pride in London Parade provides a platform for every part of London’s LGBT+ community to raise awareness of LGBT+ issues and campaign for the freedoms that will allow us to live our lives on a genuinely equal footing. It gives us a chance to be visible and speak loudly to the rest of the world about what we have achieved, how far we have come and what is still needed.",
    stages: [
      {
        stageHeading: "Trafalgar Square",
        stageSubheading: "",
        stageDescription:
          "One of the most popular and iconic areas on the day and close to where the Parade finishes. As usual there will be a mixture of famous faces and community acts from across the LGBT+ spectrum sprinkled with celebrities together with the Pride's Got Talent Finalists all under the watchful eye of Nelson's Column.",
        stageImage: "trafalgar"
      },
      {
        stageHeading: "Cabaret Stage",
        stageSubheading: "Dean Street",
        stageDescription:
          "Located in the Heart of Soho - Dean Street - no one should miss the Cabaret Stage, packed with diverse, colourful and entertaining acts from the LGBT+ Cabaret Community. This stage will also showcase the Pride's Got Talent Cabaret Finalists.",
        stageImage: "cabaret"
      },
      {
        stageHeading: "Women's / Diva Stage",
        stageSubheading: "Leicester Square",
        stageDescription:
          "Nestled in the iconic Leicester Square - this collaboration with Diva brings the best acts offered by and for the women's community. Expect high energy, fabulous talent and our female led Pride's Got Talent Acts.",
        stageImage: "diva"
      },
      {
        stageHeading: "Family Stage",
        stageSubheading: "Golden Square",
        stageDescription:
          "An intimate and fun stage, the best destination for fun and family entertainment.",
        stageImage: "family"
      },
      {
        stageHeading: "Community Village",
        stageSubheading: "Soho Square",
        stageDescription:
          "Pride in London is proud to provide a platform for charity and not for profit groups in Soho Square. Pop along to find out more about some amazing charities and not for profit groups.",
        stageImage: "community"
      }
    ]
  },
  storeVersionDialog: {
    updateAvailable: "New Update Available",
    newVersionDescription:
      "A new version of the app is available from the store. Do you want to update now?",
    askMeLater: "Ask me later",
    cancel: "Cancel",
    ok: "OK"
  },
  resetAllFilters: "Reset all filters"
};
