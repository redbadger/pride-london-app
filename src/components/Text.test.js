// @flow
import React from "react";
import { PixelRatio } from "react-native";
import { shallow } from "enzyme";
import Markdown from "react-native-easy-markdown";
import Text, { scaleFont, scaleWithFont } from "./Text";
import { lightNavyBlueColor } from "../constants/colors";

let getFontScaleSpy;
beforeEach(() => {
  getFontScaleSpy = jest.spyOn(PixelRatio, "getFontScale");
});

it("renders correctly", () => {
  const output = shallow(<Text>Some text</Text>);
  expect(output).toMatchSnapshot();
});

it("renders text based on type", () => {
  const output = shallow(<Text type="h1">Some text</Text>);

  expect(output.props().style[0]).toEqual(
    expect.objectContaining({ fontFamily: "Poppins-Bold" })
  );
});

it("renders markdown component based on flag", () => {
  const output = shallow(<Text markdown>**Some markdown**</Text>);

  expect(output).toMatchSnapshot();
});

it("does not render markdown images", () => {
  const output = shallow(
    <Text markdown>![Test](https://placehold.it/320x320.png)</Text>
  );
  const markdown = output.find(Markdown).shallow();
  expect(markdown).toMatchSnapshot();
});

it("passes custom styles down to component", () => {
  const style = { fontSize: 300 };
  const output = shallow(<Text style={style}>Some text</Text>);

  expect(output.props().style).toContainEqual(style);
});

it("renders text in blue when color is set", () => {
  const style = { color: lightNavyBlueColor };
  const output = shallow(<Text color="lightNavyBlueColor">Some text</Text>);

  expect(output.props().style).toContainEqual(style);
});

describe("scaleFont", () => {
  it("returns default font size for smaller scale", () => {
    getFontScaleSpy.mockReturnValue(0.5);
    expect(scaleFont("text", 12)).toBe(12);
  });

  it("returns default font size for default scale", () => {
    getFontScaleSpy.mockReturnValue(1);
    expect(scaleFont("text", 12)).toBe(12);
  });

  it("returns default font size for larger (but still below max) scale", () => {
    getFontScaleSpy.mockReturnValue(1.1);
    expect(scaleFont("text", 12)).toBe(12);
  });

  it("returns reduced font size for larger than max scale", () => {
    getFontScaleSpy.mockReturnValue(1.5);
    expect(scaleFont("text", 12)).toBe(10);
  });
});

describe("scaleWithFont", () => {
  it("returns reduced size for smaller scale", () => {
    getFontScaleSpy.mockReturnValue(0.5);
    expect(scaleWithFont("text", 12)).toBe(6);
  });

  it("returns default size for default scale", () => {
    getFontScaleSpy.mockReturnValue(1);
    expect(scaleWithFont("text", 12)).toBe(12);
  });

  it("returns increased size for larger (but still below max) scale", () => {
    getFontScaleSpy.mockReturnValue(1.1);
    expect(scaleWithFont("text", 12)).toBeCloseTo(13.2, 5);
  });

  it("returns max size for larger than max scale", () => {
    getFontScaleSpy.mockReturnValue(1.5);
    expect(scaleWithFont("text", 12)).toBe(15);
  });
});

afterEach(() => {
  getFontScaleSpy.mockRestore();
});
