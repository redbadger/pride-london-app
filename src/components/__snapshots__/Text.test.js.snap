// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render markdown images 1`] = `
<Component
  debug={false}
  markdownStyles={
    Object {
      "blackColor": Object {
        "color": "#111111",
      },
      "h1": Object {
        "fontFamily": "Poppins-Bold",
        "fontSize": 24,
        "includeFontPadding": false,
        "lineHeight": 28,
      },
      "h2": Object {
        "fontFamily": "Poppins-SemiBold",
        "fontSize": 18,
        "includeFontPadding": false,
        "lineHeight": 24,
      },
      "h3": Object {
        "fontFamily": "Poppins-SemiBold",
        "fontSize": 16,
        "includeFontPadding": false,
        "lineHeight": 20,
      },
      "h4": Object {
        "fontFamily": "Roboto-Medium",
        "fontSize": 16,
        "lineHeight": 24,
      },
      "lightNavyBlueColor": Object {
        "color": "#2d2f7f",
      },
      "list": Object {
        "marginBottom": 5,
        "marginHorizontal": 16,
      },
      "listItem": Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginBottom": 5,
      },
      "listItemBullet": Object {
        "alignSelf": "flex-start",
        "backgroundColor": "black",
        "borderRadius": 2,
        "height": 4,
        "marginRight": 10,
        "marginTop": 10,
        "width": 4,
      },
      "price": Object {
        "fontFamily": "Roboto-Bold",
        "fontSize": 14,
        "lineHeight": 20,
      },
      "small": Object {
        "fontFamily": "Roboto",
        "fontSize": 14,
        "lineHeight": 20,
      },
      "tabBarItem": Object {
        "fontFamily": "Poppins-Bold",
        "fontSize": 12,
        "includeFontPadding": false,
        "lineHeight": 16,
      },
      "text": Object {
        "color": "#111111",
        "fontFamily": "Roboto",
        "fontSize": 16,
        "lineHeight": 24,
      },
      "u": Object {
        "fontWeight": "bold",
      },
      "uber": Object {
        "fontFamily": "Poppins-ExtraBold",
        "fontSize": 32,
        "includeFontPadding": false,
        "lineHeight": 36,
      },
      "whiteColor": Object {
        "color": "#ffffff",
      },
      "xSmall": Object {
        "fontFamily": "Poppins-SemiBold",
        "fontSize": 12,
        "includeFontPadding": false,
        "lineHeight": 16,
      },
    }
  }
  parseInline={false}
  renderImage={[Function]}
  useDefaultStyles={true}
>
  <Component
    key="block_0"
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginBottom": 10,
      }
    }
  />
</Component>
`;

exports[`renders correctly 1`] = `
<Component
  style={
    Array [
      Object {
        "fontFamily": "Roboto",
        "fontSize": 16,
        "lineHeight": 24,
      },
      Object {
        "color": "#111111",
      },
      undefined,
    ]
  }
>
  Some text
</Component>
`;

exports[`renders markdown component based on flag 1`] = `
<Markdown
  debug={false}
  markdownStyles={
    Object {
      "blackColor": Object {
        "color": "#111111",
      },
      "h1": Object {
        "fontFamily": "Poppins-Bold",
        "fontSize": 24,
        "includeFontPadding": false,
        "lineHeight": 28,
      },
      "h2": Object {
        "fontFamily": "Poppins-SemiBold",
        "fontSize": 18,
        "includeFontPadding": false,
        "lineHeight": 24,
      },
      "h3": Object {
        "fontFamily": "Poppins-SemiBold",
        "fontSize": 16,
        "includeFontPadding": false,
        "lineHeight": 20,
      },
      "h4": Object {
        "fontFamily": "Roboto-Medium",
        "fontSize": 16,
        "lineHeight": 24,
      },
      "lightNavyBlueColor": Object {
        "color": "#2d2f7f",
      },
      "list": Object {
        "marginBottom": 5,
        "marginHorizontal": 16,
      },
      "listItem": Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginBottom": 5,
      },
      "listItemBullet": Object {
        "alignSelf": "flex-start",
        "backgroundColor": "black",
        "borderRadius": 2,
        "height": 4,
        "marginRight": 10,
        "marginTop": 10,
        "width": 4,
      },
      "price": Object {
        "fontFamily": "Roboto-Bold",
        "fontSize": 14,
        "lineHeight": 20,
      },
      "small": Object {
        "fontFamily": "Roboto",
        "fontSize": 14,
        "lineHeight": 20,
      },
      "tabBarItem": Object {
        "fontFamily": "Poppins-Bold",
        "fontSize": 12,
        "includeFontPadding": false,
        "lineHeight": 16,
      },
      "text": Object {
        "color": "#111111",
        "fontFamily": "Roboto",
        "fontSize": 16,
        "lineHeight": 24,
      },
      "u": Object {
        "fontWeight": "bold",
      },
      "uber": Object {
        "fontFamily": "Poppins-ExtraBold",
        "fontSize": 32,
        "includeFontPadding": false,
        "lineHeight": 36,
      },
      "whiteColor": Object {
        "color": "#ffffff",
      },
      "xSmall": Object {
        "fontFamily": "Poppins-SemiBold",
        "fontSize": 12,
        "includeFontPadding": false,
        "lineHeight": 16,
      },
    }
  }
  parseInline={false}
  renderImage={[Function]}
  useDefaultStyles={true}
>
  **Some markdown**
</Markdown>
`;
